activation/nipost_state.go

```diff
@@ -0,0 +1,80 @@
+package activation
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+)
+
+const (
+	challengeFilename = "NIPostChallenge"
+	builderFilename   = "NIPostBuilderState"
+	ownerReadWrite    = os.FileMode(0o600)
+)
+
+func saveNipostChallenge(dir string, ch *types.NIPostChallenge) error {
+	if _, err := os.Stat(dir); err != nil {
+		return fmt.Errorf("post dir not created: %w", err)
+	}
+	data, err := codec.Encode(ch)
+	if err != nil {
+		return fmt.Errorf("encode nipost challenge: %w", err)
+	}
+	err = os.WriteFile(filepath.Join(dir, challengeFilename), data, ownerReadWrite)
```

Comment:
as this data is somewhat critical i think we should report success only after fsync and also write checksum that is checked on the load side

write without fsync is not guaranteed to land on disk. or may land partially, in this case there will be undefined behavior on load

Reply:
thanks. done